src/nodes.coffee

```diff
@@ -961,16 +961,25 @@ exports.StringLiteral = class StringLiteral extends Literal
       @double
     }
 
+    @unquotedValueForTemplateLiteral = makeDelimitedLiteral val, {
+      delimiter: '`'
+      @double
+      escapeNewlines: no
+      includeDelimiters: no
+    }
+
+    @unquotedValueForJSX = makeDelimitedLiteral val, {
+      @double
+      escapeNewlines: no
+      includeDelimiters: no
+      escapeDelimiter: no
+    }
+
   compileNode: (o) ->
-    return [@makeCode @unquote(yes, yes)] if @jsx
+    return StringWithInterpolations.fromStringLiteral(@).compileNode o if @shouldGenerateTemplateLiteral()
```

Comment:
Given that now we're converting to a `StringWithInterpolations` (in order to generate a template literal/template literal AST) in both `StringLiteral::compileNode()` and `StringLiteral::ast()`, it'd seem cleaner to just convert it to a `StringWithInterpolations` in the constructor of `StringLiteral`

But I *believe* we don't know at construction time yet which "variant" it is (ie actual string literal vs text chunk of template literal vs JSX text)?